### ISSUE TITLE
Synchronised lists

### ISSUE THREAD

**Issue by jackrgm** [Albert221/FastShopping#24]
An important factor I consider when looking for a shopping list app is whether it supports synchronised lists for multiple users (e.g. a family), as this can be one of the major advantages an app can have over something like a text file or hardcopy.

Currently I use the [Out of Milk](https://www.outofmilk.com) app so that if my partner or I think of something for the list then we add it to the app, knowing both of us can access the shared knowledge when it's time to go shopping. I would consider FastShopping as an ideal FOSS and F-Droid alternative solution to Out of Milk if such a feature were present.

I think this feature wouldn't violate the main goal of FastShopping (it can be designed so as to be seamless and simple), but I appreciate it might be difficult to implement. Regardless, I thought this deserved a **suggestion** in case it wasn't considered yet.

**Comment by himynameisulpa** [comment]
Yeah, I was also thinking this. which was the reason I came here.

To do that, I was thinking of using nextcloud with the tasks application.

**Comment by logos88** [comment]
I also think this would be very useful, so everyone in a house could add items to a list everytime something is needed.

[ShoppingList](https://github.com/woefe/ShoppingList), included in Fdroid, stores the lists as simple text files so you can synchronize them with other devices by using applications such as Syncthing and Nextcloud. It could be a good example for an approach.

**Comment by kojid0** [comment]
Maybe creating a small nextcloud app would help here. Big benefit is that there is already a sync infrastructure available and that you can self-host it

**Comment by himynameisulpa** [comment]
> Maybe creating a small nextcloud app would help here.
No I think it's easier, if you use the task app, so you don't need to host your own nextcloud server.

**Comment by logos88** [comment]
Another open source lists app with sync is [simpletask](https://github.com/mpcjanssen/simpletask-android).

It offers three different version of the app. One without sync, other with webdav and other with nexcloud. For instance, here is the webdav version:

[https://f-droid.org/es/packages/nl.mpcjanssen.simpletask.webdav/](https://f-droid.org/es/packages/nl.mpcjanssen.simpletask.webdav/)
[https://github.com/mpcjanssen/simpletask-android/issues/1040](https://github.com/mpcjanssen/simpletask-android/issues/1040)

**Comment by Chris9191** [comment]
I recently discovered the app and would also support this feature. I would suggest to check other possibilities of synchronization besides the ones already mentioned. For example, via wifi or Bluetooth. Or simply sending a file via Messenger, so that the partner can simply import this and has the list available.

**Comment by WST1811** [comment]
I'm looking for a Shopping List App with synchronization WebDAV/Nextcloud too.

**Comment by NohWayJose** [comment]
Would a Next-cloud solution be OwnCloud compatible (I have OC, not NC but I believe NC is a fork of OC and there's a debate about which is better - I don't know the answer to that, just that I don't want to have to port all my clients across)

**Comment by kojid0** [comment]
There is Nextcloud Cospend App (https://github.com/eneiluj/cospend-nc) which takes care of syncing bills between Nextcloud Server and an Android/iOS app. Maybe you can take the code base and adapt for FastShopping.

**Comment by SimonBrodtmann** [comment]
This looks like a really fine app. I like the looks and the simplicity. But I have the same problem as the other commenters. I'm not alone so I need a solution with synchronization. Currently we use Out of Milk but the app was sold to an american company that is not active in europe and they don't plan to make the app available to europeans.

I'm not a fan of the proposed nextcloud solution. Not because nextcloud is bad (I'm using it myself and I love it), but because it's a very heavy dependency for a very simple problem. Nobody would be happy to install nextcloud just to make this app run with synchronization. Installing a simple web backend into an existing webserver is a matter of minutes.

tl;dr
I could create a simple PHP backend for a self-hosted solution. Personally I like playing around with NodeJS more but I think PHP is the more compatible solution.
Who would want to do the integration into the app?

**Comment by brunoais** [comment]
> I could create a simple PHP backend for a self-hosted solution. Personally I like playing around with NodeJS more but I think PHP is the more compatible solution.
> Who would want to do the integration into the app?

I'd rather just have WebDav. At least, for now. It's generic, already available in wide range of libraries both for servers and clients and it's available free of charge and out-of-the-box with an apache installation with minimal configuration necessary.

**Comment by jackrgm** [comment]
Closing issue due to inactivity.

For others still looking for a solution, I've been using CalDAV apps with a 'Shopping' task collection, which gets the job done. It also means one less app if you use it for other tasks etc. too.
